src/libcommonserver/io/iohelper.cpp

```diff
@@ -98,6 +98,9 @@ IoError IoHelper::posixError2ioError(int error) noexcept {
             return IoErrorInvalidArgument;
         case ENAMETOOLONG:
             return IoErrorFileNameTooLong;
+#ifdef _WIN32
+        case ESRCH:
```

Comment:
Strangely std::filesystem::permission return ec ESRCH instead of ENOENT when the file does not exist on Windows.